cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicSystemHttpServiceContext.java

```diff
@@ -152,7 +155,14 @@ public void run(TxRunnable runnable) throws TransactionException {
     String systemAppParam = GSON.toJson(runnableTaskRequest);
```

Comment:
Seems like this method should check if task worker is enabled?

Reply:
System apps will be able to check if remote task is enabled before calling this. Will it help to add additional check here as a safe guard?

Reply:
If the caller forget to check before calling this method, what will happen?

Reply:
Makes sense. I can return a sensible error message here if it is disabled.

Reply:
Done